### DISCUSSION TITLE
Allergens - Its posible to add if the recipe contains any of the 14 allegens

### DISCUSSION THREAD

**Discussion by ikanpos** [mealie-recipes/mealie#1239] [Feature Request]
I'm from Spain and i am very happy with this application. I am very pleasantly surprised with your application. In fact, I have already recommended it to several cooking teachers. Here in Spain we use a table of allergens with their icon to be able to fix them in the menus. In the USA I don't know how things are going, the last time I was there (2 years ago) I didn't see anything in the restaurants I was at. 
It would be fabulous to be able to integrate if it contains any of the 14 allergens that are required here. In fact it would not be bad to also be able to indicate if it is Vegan, Kosher, Halal... Basically a checkbox. So that we are not overlooked when we reach for the recipes. 
Congratulations and thank you very much for this magnificent "mealie". 
I leave a link so you can understand what I'm talking about. 
it's in Spanish, at the end there are 14 icons (1 for each allergen).
https://www.chefejecutivo.com/iconos-simbolos-alergenos-restaurantes/
Thank you so much

**Comment by geoffbon** [comment]
Also: https://www.foodprotection.org/resources/food-allergen-icons/
(assuming non-commercial use)

**Comment by hay-kot** [comment]
I like this idea, could you use the feature request for here and update your post? 

- https://github.com/hay-kot/mealie/issues/317

I can see it being useful to specify allergens on ingredients and those show up in the viewer somehow. It would be helpful is you or anyone else had some suggestions on how to display that visually.

**Comment by jarnohissink** [comment]
@hay-kot Maybe somthing like this - i only have edited in a few icons
![allergenen mealie](https://github.com/mealie-recipes/mealie/assets/5670564/4df3b73a-9a86-4b88-bef4-af9545b7c787)

You can make the icons switchable by clicking on them in edit mode
hover over them to give more information in recipe mode



I have added a link to the icons i used:
https://codepen.io/jarno-hissink/pen/YzMLONM
[icomoon.zip](https://github.com/mealie-recipes/mealie/files/14931909/icomoon.zip)

**Comment by jarnohissink** [comment]
Maybe it is even possible to trigger the icons when you add a ingedient like "milk" or "yoghurt" it troggle the dairy icon

**Comment by CloCkWeRX** [comment]
Possibly link to openfoodfacts' allergen lists would surface some of this:

https://world.openfoodfacts.org/allergens
https://github.com/openfoodfacts/openfoodfacts-server/blob/main/taxonomies/allergens.txt

Or reference via wikidata ids
